Mathlib/Topology/LocalAtTarget.lean

```diff
@@ -64,16 +64,13 @@ alias ClosedEmbedding.restrictPreimage := Set.restrictPreimage_closedEmbedding
 #align closed_embedding.restrict_preimage ClosedEmbedding.restrictPreimage
 
 theorem Set.restrictPreimage_isClosedMap (s : Set β) (H : IsClosedMap f) :
-    IsClosedMap (s.restrictPreimage f) := by
-  rintro t ⟨u, hu, e⟩
-  refine' ⟨⟨_, (H _ (IsOpen.isClosed_compl hu)).1, _⟩⟩
-  rw [← (congr_arg HasCompl.compl e).trans (compl_compl t)]
-  simp only [Set.preimage_compl, compl_inj_iff]
-  ext ⟨x, hx⟩
-  suffices (∃ y, y ∉ u ∧ f y = x) ↔ ∃ y, y ∉ u ∧ f y ∈ s ∧ f y = x by
-    simpa [Set.restrictPreimage, ← Subtype.coe_inj]
-  exact ⟨fun ⟨a, b, c⟩ => ⟨a, b, c.symm ▸ hx, c⟩, fun ⟨a, b, _, c⟩ => ⟨a, b, c⟩⟩
-#align set.restrict_preimage_is_closed_map Set.restrictPreimage_isClosedMap
+    IsClosedMap (s.restrictPreimage f) :=
+  fun t ⟨u, hu, e⟩ =>  ⟨(f '' uᶜ)ᶜ, isOpen_compl_iff.2 (H uᶜ (IsOpen.isClosed_compl hu)),
+    by simp [e, ← image_restrictPreimage]⟩
+
+theorem Set.restrictPreimage_isOpenMap (s : Set β) (H : IsOpenMap f) :
+    IsOpenMap (s.restrictPreimage f) :=
+  fun t ⟨u, hu, e⟩ => ⟨f '' u, H u hu, by rw [← e, image_restrictPreimage]⟩
```

Comment:
If you want the proof to look exactly the same as the closed case you can use [isOpen_induced_iff](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Order.html#isOpen_induced_iff) instead of relying on the fact that it's true by definition. But really you don't have to, do as you wish.